File: core/src/galenscovell/hinterstar/ui/components/gamescreen/hud/CrewPanel.scala
package galenscovell.hinterstar.ui.components.gamescreen.hud

import com.badlogic.gdx.scenes.scene2d.ui._
import galenscovell.hinterstar.things.entities.Crewmate
import galenscovell.hinterstar.ui.components.gamescreen.stages.HudStage
import galenscovell.hinterstar.util._


class CrewPanel(hudStage: HudStage) extends Table {
  private val contentTable: Table = new Table

  construct()



  private def construct(): Unit = {
    refresh()
    this.add(contentTable).expand.left.bottom
  }

  def refresh(): Unit = {
    contentTable.clear()

    for (crewmate: Crewmate <- PlayerData.getCrew) {
      contentTable.add(crewmate.getCrewBox).width(120).height(60).left.padRight(12)
    }
  }
}

File: core/src/galenscovell/hinterstar/ui/components/gamescreen/hud/CrewPanel.kt
package galenscovell.hinterstar.ui.components.gamescreen.hud

import com.badlogic.gdx.scenes.scene2d.ui.Table
import galenscovell.hinterstar.things.entities.Crewmate
import galenscovell.hinterstar.ui.components.gamescreen.stages.InterfaceStage
import galenscovell.hinterstar.util.PlayerData


class CrewPanel(interfaceStage: InterfaceStage) : Table() {
    private val contentTable: Table = Table()

    init {
        refresh()
        this.add(contentTable).expand().left().bottom()
    }



    fun refresh(): Unit {
        contentTable.clear()

        for (crewmate: Crewmate in PlayerData.getCrew()) {
            contentTable.add(crewmate.getCrewBox()).width(120f).height(60f).left().padRight(12f)
            crewmate.setAssignmentIcon()
        }
    }
}
